datalad/tests/testrepos/items.py

```diff
@@ -0,0 +1,1396 @@
+        assert(self._state_worktree in states)
+
+        if self.is_untracked:
+            assert(self.is_staged is False)
+            assert(self.is_modified is False)
+            assert(self.is_clean is False)
+            assert(self.is_locked is None)
+            assert(self.is_unlocked is None)
+
+            assert(not self.commits)
+            assert(self.annexed is False)
+            assert(self.annex_key is None)
+            assert(self.content_available is None)
+
+        if self.is_modified or self.is_staged:
+            assert(self.is_clean is False)
+            assert(self.is_untracked is False)
+
+        if self.is_clean:
+            assert(self.is_untracked is False)
+            assert(self.is_staged is False)
+            assert(self.is_modified is False)
+
+        # if it's neither untracked nor staged as a new file, there needs to be
+        # a commit and vice versa:
+        assert((self._state_index in [ItemFile.UNMODIFIED,
+                                 ItemFile.MODIFIED,
+                                 ItemFile.DELETED,
+                                 ItemFile.RENAMED,
+                                 ItemFile.COPIED,
+                                 ItemFile.TYPECHANGED,
+                                 ItemFile.UPDATED_BUT_UNMERGED]
+                and isinstance(self.commits, list)
+                ) or
+               (self._state_index in [ItemFile.UNTRACKED, ItemFile.ADDED]
+                and not self.commits
+                )
+               )
+
+        if self.is_locked or self.is_unlocked \
+                or self.annex_key or self.content_available:
+            assert self.annexed
+
+        if self.annexed:
+            assert(self.is_untracked is False)
+            assert_is_instance(self.annex_key, string_types)
+            assert_is_instance(self.content_available, bool)
+            assert_is_instance(self.commits, list)
+
+        # Note: `None` if "locked/unlocked" isn't a valid concept (plain git,
+        # untracked, direct mode):
+        assert(not(self.is_locked and self.is_unlocked))
+        assert(not(self.is_locked is False and self.is_unlocked is False))
+
+        # physical appearance:
+
+        assert(lexists(self.path))
+
+        if self.content_available or not self.annexed:
+            with open(self.path, 'r') as f:
+                content_from_disc = f.read()
+            eq_(content_from_disc, self.content)
+
+        if self.is_locked:
+            assert_raises(EnvironmentError, open, self.path, 'w')
+
+        # TODO: if not self.is_untracked: git knows it
+        # TODO: if self.annexed: annex finds it and provides a key
+        # TODO: compare commits
+        # TODO: - state? This might need annex-proxy and ignore-submodules, but
+        #         since we have a certain file to specify it might work
+        #       - see corresponding note in ItemRepo
+
+
+@auto_repr
+class ItemInfoFile(ItemFile):
```

Comment:
invaluable concern:  use "Item" as a suffix (if at all) not as a prefix, to stay inline with other places (AnnexRepo, etc)

Reply:
here is an example for attrs
```python
>>> import attr
>>> @attr.s
... class C(object):
...     x = attr.ib(default=42)
...     y = attr.ib(default=attr.Factory(list))
```

so we could do just that `from testrepos import items`, and use `items.File`, or if really valuing our bytes, `from testrepos import items as it;  it.File`